### DISCUSSION TITLE
assign / for_window based on Title

### DISCUSSION THREAD

**Discussion by MillingtonAllen** [i3/i3#4977] [Q&A] [answered]
I want i3 to automatically assing a Konsole window of a specific title (to be launched at startup) to workspace 3.

`exec konsole --workdir ~/cs242/`

I ran xprop and it outputs:

```
XdndAware(ATOM) = BITMAP
WM_NAME(STRING) = "cs242 : bash"
_NET_WM_NAME(UTF8_STRING) = "cs242 : bash — Konsole"
_MOTIF_WM_HINTS(_MOTIF_WM_HINTS) = 0x3, 0x3e, 0x7e, 0x0, 0x0
_NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_NORMAL
_XEMBED_INFO(_XEMBED_INFO) = 0x0, 0x1
WM_CLIENT_LEADER(WINDOW): window id # 0x3a00009
WM_HINTS(WM_HINTS):
                Client accepts input or input focus: True
                window id # of group leader: 0x3a00009
WM_CLIENT_MACHINE(STRING) = "allen-desktop"
_NET_WM_PID(CARDINAL) = 11073
_NET_WM_SYNC_REQUEST_COUNTER(CARDINAL) = 60817416
WM_CLASS(STRING) = "konsole", "konsole"
WM_PROTOCOLS(ATOM): protocols  WM_DELETE_WINDOW, WM_TAKE_FOCUS, _NET_WM_PING, _NET_WM_SYNC_REQUEST
WM_NORMAL_HINTS(WM_SIZE_HINTS):
                user specified location: 1924, 30
                user specified size: 1912 by 2090
                program specified minimum size: 8 by 52
                program specified resize increment: 2 by 2
                program specified base size: 0 by 0
                window gravity: Static
```

I tried `assign [title="cs242 : bash — Konsole"] → $ws3`

I have tried both the _NET_WM_NAME and WM_NAME titles. I have tried assign instead of for_window. I thought maybe Konsole was changing it's title on launch, so I tried with for_window instead of assign and it didn't work either. I tried various substrings of the title as well and had no luck.  Any suggestions?

**Comment by Airblader** [comment, marked as answer]
Just for triaging purposes, let's try this:

```
for_window [title="cs242"] move window to workspace $ws3
```

If it doesn't work, please prepare a log file as described [here](https://i3wm.org/docs/debugging.html).

**Reply by MillingtonAllen** [reply]
Looks like using the "move window to workspace" works with `cs242` as well as  `cs242 : bash — Konsole` and `cs242 : bash`.

I'm running version i3 version 4.19.1-non-git in Ubuntu 21.10.  I've been holding off upgrading to Ubuntu 22.04 due to current projects and not wanting to risk downtime / incompatibilities.

I'm not sure why the `→` character is throwing things off.  I'm guessing the arrow character doesn't like the colon ':' character or 'title'.  I used the arrow character in all other places and it was fine. (example: 
`assign [class="Eclipse"] → $ws1` works as expected). 

Here is a debug log.
https://logs.i3wm.org/logs/6271712832258048.bz2

**Reply by Airblader** [reply]
I'm confused as to why that is working when you said you had tried it before:

> I have tried assign instead of for_window. I thought maybe Konsole was changing it's title on launch, so I tried with for_window instead of assign and it didn't work either.

There is no `→` when using `for_window`, and for `assign` it's optional.

**Reply by MillingtonAllen** [reply]
I had the wrong syntax on `for_window` originally.  I misunderstood the arrow as `move to` rather than a decorative _beautiful arrow_.

Thanks for the help.
